src/gc/gc.d

```diff
@@ -2329,27 +2321,82 @@ struct Gcx
         return 1;
     }
 
-    /**
-     * Mark overload for initial mark() call.
-     */
-    void mark(void *pbot, void *ptop) nothrow
+    static struct ToScanStack
     {
-        mark(pbot, ptop, MAX_MARK_RECURSIONS);
+    nothrow:
+        @disable this(this);
+
+        void reset()
+        {
+            _length = 0;
+            os_mem_unmap(_p, _cap * Range.sizeof);
+            _p = null;
+            _cap = 0;
+        }
+
+        void push(Range rng)
+        {
+            if (_length + 1 >= _cap) grow();
+            _p[_length++] = rng;
+        }
+
+        Range pop()
+        in { assert(!empty); }
+        body
+        {
+            return _p[--_length];
+        }
+
+        ref inout(Range) opIndex(size_t idx) inout
+        in { assert(idx < _length); }
+        body
+        {
+            return _p[idx];
+        }
+
+        @property size_t length() const { return _length; }
+        @property bool empty() const { return !length; }
+
+    private:
+        void grow()
+        {
+            immutable ncap = _cap ? 2 * _cap : PAGESIZE / Range.sizeof;
```

Comment:
OK done, won't hurt anyone. From my understanding 64kB granularity of address space doesn't mean it needs to wire the complete 64kB, so only using 4kB would save resources, not that it would matter here.